### PR TITLE
AHAB support

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

File: security/imx_ahab/sign-file.sh [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    done
+}
+
+parse_args "$@"
+
+if [ "${DISPLAY_USAGE}" = "1" ]; then
+    echo "usage for: ${0}"
+    echo "  --cst: set cst binary path/filename   [default: ${CST_BINARY}]"
+    echo "  --csf-template: set CSF template file [default: ${CSF_TEMPLATE}]"
+    echo "  --spl: SPL binary to sign             [--spl or --m4app is required]"
```
Comment (vanmaegima) [inline security/imx_ahab/sign-file.sh:87]:
```suggestion
    echo "  --spl: SPL binary to sign             [--spl is required]"
```

Comment (igoropaniuk) [reply security/imx_ahab/sign-file.sh]: fixed, thanks

**Review by vanmaegima — APPROVED** [review]
LGTM

**Review by MrCry0 — APPROVED** [review]
LGTM

**Review by Tim-Anderson — COMMENTED** [review]
Are these the same keys as was used on the hab4 implementation? if not, why a new set of keys?  We have tried to have a single set of keys to sign development boards to be consistent so a closed board could be shared accross the company.

**Comment by igoropaniuk** [comment]
> Are these the same keys as was used on the hab4 implementation? if not, why a new set of keys? We have tried to have a single set of keys to sign development boards to be consistent so a closed board could be shared across the company

no, they are not the same. there are different scripts for generating keys for HABv4 vs AHAB (`ahab_pki_tree.sh` vs `hab4_pki_tree.sh`), and based on their contents AHAB has a bit different PKI requirements, for example HABv4 can support 2 subordinate keys (CSF vs IMG), however AHAB only one (SGK). AHAB doesn't support 1024 RSA key length (this is what I saw from `ahab_pki_tree.sh` sources),  (just an assumption) if you generate 1024 key/pair (with `hab4_pki_tree.sh`) and sign with it boot binary, iMX8QM SECO ROM likely won't accept it.

So instead of taking existing keys and making them aligned with AHAB requirements, I decided to do everything "by book" (as explained in the AHAB manual) to avoid any mistakes.

if user want's to generate his own key pairs/certificates, he will definitely use `ahab_pki_tree.sh` script from CST package and that's it, so this is what I did.
 
Additional details:
https://source.codeaurora.org/external/imx/uboot-imx/tree/doc/imx/habv4/introduction_habv4.txt?h=imx_v2020.04_5.4.70_2.3.0
https://source.codeaurora.org/external/imx/uboot-imx/tree/doc/imx/ahab/introduction_ahab.txt?h=imx_v2020.04_5.4.70_2.3.0

**Review by angolini — APPROVED** [review]

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**